### PR TITLE
explain for connection: fix the compatibility with MySQL

### PR DESCRIPTION
**Pull request by Oreoxmt**
### What is changed, added or deleted? (Required)

`EXPLAIN FOR CONNECTION` is used to get the execution plan of the currently executed SQL query or the last executed SQL query in a connection.

Fix the compatibility to make it consistent with https://github.com/pingcap/tidb/pull/10030

```
Differences with MySQL EXPLAIN FOR

If one connection is sleeping, query for query plan for that connection will get nothing in MySQL. However, in this PR, query play for last query will be returned.
Side effect of 1., EXPLAIN FOR CONNECTION <id of this connection> will get error in MySQL, but will get query plan of last query in this PR.
If one without ProcessPriv queries plan for another user's connection, MySQL won't return error if that connection does not running any queries, however, in this PR, an error will always returns, even if that connection does not run any queries yet.
```

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/12536
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by xhebox — APPROVED** [review]

File: sql-statements/sql-statement-explain.md [hunk 1]
```diff
@@ -343,7 +343,6 @@ EXPLAIN FORMAT = "tidb_json" SELECT id FROM t WHERE a = 1;
 
 `EXPLAIN FOR CONNECTION` 用于获得一个连接中当前正在执行 SQL 的执行计划或者是最后执行 SQL 的执行计划，其输出格式与 `EXPLAIN` 完全一致。但 TiDB 中的实现与 MySQL 不同，除了输出格式之外，还有以下区别：
 
-- MySQL 返回的是**正在执行**的查询计划，而 TiDB 返回的是**最后执行**的查询计划。
 - MySQL 的文档中指出，MySQL 要求登录用户与被查询的连接相同，或者拥有 `PROCESS` 权限，而 TiDB 则要求登录用户与被查询的连接相同，或者拥有 `SUPER` 权限。
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-explain.md:346]:
```suggestion
- 如果连接处于睡眠状态，MySQL 返回为空，而 TiDB 返回的是最后执行的执行计划。
- 如果获取当前会话连接的执行计划，MySQL 会报错，而 TiDB 会正常返回。
- MySQL 的文档中指出，MySQL 要求登录用户与被查询的连接相同，或者拥有 `PROCESS` 权限，而 TiDB 则要求登录用户与被查询的连接相同，或者拥有 `SUPER` 权限。
```
@xhebox PTAL

Comment (xhebox) [reply sql-statements/sql-statement-explain.md:346]: LGTM

File: sql-statements/sql-statement-explain.md [hunk 2]
```diff
@@ -343,7 +343,8 @@ EXPLAIN FORMAT = "tidb_json" SELECT id FROM t WHERE a = 1;
 
 `EXPLAIN FOR CONNECTION` 用于获得一个连接中当前正在执行 SQL 的执行计划或者是最后执行 SQL 的执行计划，其输出格式与 `EXPLAIN` 完全一致。但 TiDB 中的实现与 MySQL 不同，除了输出格式之外，还有以下区别：
```
Comment (hfxsd) [inline sql-statements/sql-statement-explain.md:344]: 这句话有歧义，前面说 “其输出格式与 `EXPLAIN` 完全一致”， 后面说“除了输出格式之外，还有以下区别”，似乎输出格式又不同。需要更新描述。

Comment (Oreoxmt) [reply sql-statements/sql-statement-explain.md:344]: 这里是指 TiDB `EXPLAIN FOR CONNECTION` 与 TiDB `EXPLAIN` 的输出格式是一致的。但是 TiDB `EXPLAIN` 与 MySQL `EXPLAIN` 在输出格式上存在差异，上面兼容性部分已经说明了。

```markdown
## MySQL 兼容性
* `EXPLAIN` 的格式和 TiDB 中潜在的执行计划都与 MySQL 有很大不同。
* TiDB 不支持 `FORMAT=JSON` 或 `FORMAT=TREE` 选项。
* TiDB 支持的 `FORMAT=tidb_json` 是对当前默认 `EXPLAIN` 格式的 JSON 编码，与 MySQL 的 `FORMAT=JSON` 结果的格式、字段信息都不同。
```

**Review by hfxsd — APPROVED** [review]

**Pull request merged**